icu4c/source/common/cjmlbe.h

```diff
@@ -0,0 +1,129 @@
+     */
+     int32_t divideUpDictionaryRange(UText *inText, int32_t rangeStart, int32_t rangeEnd,
+                                     UVector32 &foundBreaks, const UnicodeString &inString,
+                                     const LocalPointer<UVector32> &inputMap,
+                                     UErrorCode &status) const;
+
+private:
+
+    /**
+     * Load the machine learning's model file.
+     *
+     * @param error Information on any errors encountered.
+     */
+    void loadMLmodel(UErrorCode &error);
+
+    /**
+     * Get the unicode block index of the unicode character.
+     *
+     * @param ch A unicode character.
+     * @param error Information on any errors encountered.
+     * @return The unicode block index
+     */
+     UnicodeString getUnicodeBlockIndex(UChar32 ch, UErrorCode &status) const;
+
+    /**
+     * Initiate the element list from the input string.
+     *
+     * @param inString A input string to be segmented
+     * @param elementList A vector to store the first six characters and their unicode block indexes.
+     * @param status A vector to store the first six characters and their unicode block indexes.
+     * @param status Information on any errors encountered.
+     */
+    void initElementList(const UnicodeString &inString, UVector &elementList,
+                         UErrorCode &status) const;
+
+    /**
+     * Evaluate whether the index is a potential breakpoint.
+     *
+     * @param w1 The character, 3 characters before the index.
+     * @param w2 The character, 2 characters before the index.
+     * @param w3 The character, right before the index.
+     * @param w4 The character, right after the index.
+     * @param w5 The character, 2 characters after the index.
+     * @param w6 The character, 3 characters after the index.
+     * @param b1 The unicode block index of w1.
+     * @param b2 The unicode block index of w2.
+     * @param b3 The unicode block index of w3.
+     * @param b4 The unicode block index of w4.
+     * @param b5 The unicode block index of w5.
+     * @param b6 The unicode block index of w6.
+     * @param index The breakpoint index to be evaluated.
+     * @param numBreaks The accumulated number of breakpoints.
+     * @param boundary A vector including the index of the breakpoint.
+     * @param status Information on any errors encountered.
+     */
+    void evaluateBreakpoint(UChar32 w1, UChar32 w2, UChar32 w3, UChar32 w4, UChar32 w5,
+                            UChar32 w6, const UnicodeString &b1, const UnicodeString &b2,
+                            const UnicodeString &b3, const UnicodeString &b4,
+                            const UnicodeString &b5, const UnicodeString &b6,
+                            int32_t index, int32_t &numBreaks, UVector32 &boundary,
+                            UErrorCode &status) const;
+
+    int32_t negativeSum;
+    UnicodeSet fDigitOrOpenPunctuationOrAlphabetSet;
+    UnicodeSet fClosePunctuationSet;
+    Hashtable fModel;
```

Comment:
swap the order of fModel and fUnicodeBlock since in your constructor you are dealing with fUnicodeBlock before fModel

Reply:
Remove fUnicodeBlock.